some compile warnings

```
pio_darray.c: In function ‘PIOc_read_darray’:
pio_darray.c:872:9: warning: unused variable ‘pio_type’ [-Wunused-variable]
  872 |     int pio_type;
      |         ^~~~~~~~
pio_darray.c:871:11: warning: unused variable ‘fillvalue’ [-Wunused-variable]
  871 |     void *fillvalue = NULL;
      |           ^~~~~~~~~
pioc.c: In function ‘PIOc_init_async_from_comms’:
pioc.c:2069:10: warning: variable ‘in_io’ set but not used [-Wunused-but-set-variable]
 2069 |     bool in_io = false;
      |          ^~~~~
pioc.c:2050:9: warning: unused variable ‘mpierr’ [-Wunused-variable]
 2050 |     int mpierr;           /* Return code from MPI functions. */
      |         ^~~~~~
```

And...

```
pio_msg.c: In function ‘set_loglevel_handler’:
pio_msg.c:2599:5: warning: ‘level’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 2599 |     PIOc_set_global_log_level(iosysid, level);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

--- a/src/clib/pio_darray.c
+++ b/src/clib/pio_darray.c
@@ -868,8 +868,6 @@ PIOc_read_darray(int ncid, int varid, int ioid, PIO_Offset arraylen,
     size_t rlen = 0;       /* the length of data in iobuf. */
     void *tmparray;        /* unsorted copy of array buf if required */
     int mpierr = MPI_SUCCESS, mpierr2;  /* Return code from MPI function calls. */
-    void *fillvalue = NULL;
-    int pio_type;
     int ierr;              /* Return code. */
 
 #ifdef USE_MPE

--- a/src/clib/pioc.c
+++ b/src/clib/pioc.c
@@ -924,7 +924,9 @@ PIOc_InitDecomp_bc(int iosysid, int pio_type, int ndims, const int *gdimlen,
  *
  * @param comp_comm the MPI_Comm of the compute tasks.
  * @param num_iotasks the number of io tasks to use.
- * @param stride the offset between io tasks in the comp_comm.
+ * @param stride the offset between io tasks in the comp_comm. The mod
+ * operator is used when computing the IO tasks with the formula:
+ * <pre>ios->ioranks[i] = (base + i * ustride) % ios->num_comptasks</pre>.
  * @param base the comp_comm index of the first io task.
  * @param rearr the rearranger to use by default, this may be
  * overriden in the PIO_init_decomp(). The rearranger is not used

--- a/src/clib/pioc.c
+++ b/src/clib/pioc.c
@@ -2041,14 +2041,18 @@ PIOc_init_async(MPI_Comm world, int num_io_procs, int *io_proc_list,
  * @author Jim Edwards
  */
 int
-PIOc_init_async_from_comms(MPI_Comm world, int component_count, MPI_Comm *comp_comm, MPI_Comm io_comm, int rearranger, int *iosysidp)
+PIOc_init_async_from_comms(MPI_Comm world, int component_count, MPI_Comm *comp_comm,
+                           MPI_Comm io_comm, int rearranger, int *iosysidp)
 {
     int my_rank;          /* Rank of this task. */
     int **my_proc_list;   /* Array of arrays of procs for comp components. */
     int *io_proc_list; /* List of processors in IO component. */
     int *num_procs_per_comp; /* List of number of tasks in each component */
-    int mpierr;           /* Return code from MPI functions. */
+    int num_io_procs = 0;
     int ret;              /* Return code. */
+#ifdef USE_MPE
+    bool in_io = false;
+#endif /* USE_MPE */
 
 #ifdef USE_MPE
     pio_start_mpe_log(INIT);
@@ -2065,10 +2069,11 @@ PIOc_init_async_from_comms(MPI_Comm world, int component_count, MPI_Comm *comp_c
     PLOG((1, "PIOc_init_async_from_comms component_count = %d", component_count));
 
     /* Get num_io_procs from io_comm, share with world */
-    int num_io_procs = 0;
-    bool in_io = false;
-    if(io_comm != MPI_COMM_NULL) {
+    if (io_comm != MPI_COMM_NULL)
+    {
+#ifdef USE_MPE
         in_io = true;
+#endif /* USE_MPE */
         if ((ret = MPI_Comm_size(io_comm, &num_io_procs)))
             return check_mpi(NULL, NULL, ret, __FILE__, __LINE__);
     }
@@ -2077,7 +2082,8 @@ PIOc_init_async_from_comms(MPI_Comm world, int component_count, MPI_Comm *comp_c
 
     /* Get io_proc_list from io_comm, share with world */
     io_proc_list = (int*) calloc(num_io_procs, sizeof(int));
-    if(io_comm != MPI_COMM_NULL) {
+    if (io_comm != MPI_COMM_NULL)
+    {
         int my_io_rank;
         if ((ret = MPI_Comm_rank(io_comm, &my_io_rank)))
             return check_mpi(NULL, NULL, ret, __FILE__, __LINE__);

--- a/src/flib/piolib_mod.F90
+++ b/src/flib/piolib_mod.F90
@@ -967,7 +967,8 @@ contains
   !! @param iosystem a derived type which can be used in subsequent
   !! pio operations (defined in PIO_types).
   !! @param base @em optional argument can be used to offset the first
-  !! io task - default base is task 1.
+  !! io task. Since this is an MPI task number, it is zero-based (the
+  !! first task is 0). The default base is task 1.
   !! @param rearr_opts the rearranger options.
   !! @author Jim Edwards
   !<

--- a/src/flib/piolib_mod.F90
+++ b/src/flib/piolib_mod.F90
@@ -968,7 +968,7 @@ contains
   !! pio operations (defined in PIO_types).
   !! @param base @em optional argument can be used to offset the first
   !! io task. Since this is an MPI task number, it is zero-based (the
-  !! first task is 0). The default base is task 1.
+  !! first task is 0). The default base is task 0.
   !! @param rearr_opts the rearranger options.
   !! @author Jim Edwards
   !<

--- a/tests/unit/driver.F90
+++ b/tests/unit/driver.F90
@@ -102,7 +102,7 @@ Program pio_unit_test_driver
   call MPI_Bcast(stride,1,MPI_INTEGER,0,MPI_COMM_WORLD,ierr)
   niotasks = ntasks/stride
 
-  ! Set up PIO
+  ! Set up PIO. Use a base of 1 because task 0 is already busy.
   call PIO_init(my_rank,        & ! MPI rank
        MPI_COMM_WORLD, & ! MPI communicator
        niotasks,       & ! Number of iotasks (ntasks/stride)

